src/equisolve/numpy/_selection.py

```diff
@@ -0,0 +1,85 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+#
+# Copyright (c) 2023 Authors and contributors
+# (see the AUTHORS.rst file for the full list of names)
+#
+# Released under the BSD 3-Clause "New" or "Revised" License
+# SPDX-License-Identifier: BSD-3-Clause
+
+from equistore import TensorMap, slice_block
+from skmatter._selection import GreedySelector
+
+
+class GreedySelector:
+    """Wrapper for :py:class:`skmatter._selection.GreedySelector` for a
+    :py:class:equistore.TensorMap`.
+
+    Create a selector for each block. The selection will be done based
+    the values of each :py:class:`TensorBlock`.
+    """
+
+    def __init__(
+        self, selector: GreedySelector, selection_type: str, **selector_arguments
+    ) -> None:
+        self.selection_type = selection_type
+        self.selector_arguments = selector_arguments
+        self.selector = selector
```

Comment:
```suggestion
        self.selector_class = selector
```

we are not storing the instance here, right?

Reply:
Not an instance, but the class.
The instance per class is also saved.

Reply:
I think the rename @Luthaf suggested would be also good.

According to the typing it should be an instance, but it is a class. I think the correct typing would be a `type` or to be more specific Type[GreedySelector]
```python
from typing import Type
...
selector: Type[GreedySelector]
```

This is how mypy deals with it
```python
from typing import Type

class A: pass
class B(A): pass
class C: pass

def fooA() -> Type[A]:
    return A # works

def fooB() -> Type[A]:
    return B # works

def fooC() -> Type[A]:
    return C # cause error

```
Even though for GreedySelector some stuff is weird
```python
from skmatter._selection import GreedySelector
from skmatter.feature_selection import FPS
from typing import Type

def foo_correct() -> Type[GreedySelector]:
    return FPS # works

def foo_why_does_this_work() -> Type[GreedySelector]:
    return type(5) # works

def foo_wrong() -> Type[GreedySelector]:
    return 5 # error
```
It seems to default to `Type[Any]`, because there is no typing support in skmatter? Anyway we can think about this another time.

Reply:
Yes I did the renaming!